File: frontend/src/data/board.ts
export type PropertyGroup =
    | 'Brown'
    | 'LightBlue'
    | 'Pink'
    | 'Orange'
    | 'Red'
    | 'Yellow'
    | 'Green'
    | 'DarkBlue'
    | 'Railroad'
    | 'Utility'
    | 'Special' // For GO, Jail, etc.
    | 'Tax'
    | 'Chance'
    | 'Chest';

export interface TileInfo {
    id: number;
    name: string;
    type: 'Property' | 'Corner' | 'Tax' | 'Chance' | 'Chest' | 'Railroad' | 'Utility';
    price?: number;
    rent?: number;
    group?: PropertyGroup;
}

export const TILES: TileInfo[] = [
    // Bottom Row (0-10)
    { id: 0, name: "Wisuda (GO)", type: 'Corner', group: 'Special' },
    { id: 1, name: "Matematika", type: 'Property', price: 60000, rent: 2000, group: 'Brown' },
    { id: 2, name: "BEM", type: 'Chest', group: 'Chest' },
    { id: 3, name: "Fisika", type: 'Property', price: 60000, rent: 4000, group: 'Brown' },
    { id: 4, name: "Bayar UKT", type: 'Tax', price: 200000, group: 'Tax' },
    { id: 5, name: "Stasiun UI", type: 'Railroad', price: 200000, rent: 25000, group: 'Railroad' },
    { id: 6, name: "Sastra Inggris", type: 'Property', price: 100000, rent: 6000, group: 'LightBlue' },
    { id: 7, name: "SIAK-NG", type: 'Chance', group: 'Chance' },
    { id: 8, name: "Arkeologi", type: 'Property', price: 100000, rent: 6000, group: 'LightBlue' },
    { id: 9, name: "Filsafat", type: 'Property', price: 120000, rent: 8000, group: 'LightBlue' },
    { id: 10, name: "Skorsing", type: 'Corner', group: 'Special' },

    // Left Column (11-19)
    { id: 11, name: "Ilmu Komunikasi", type: 'Property', price: 140000, rent: 10000, group: 'Pink' },
    { id: 12, name: "Perpustakaan UI", type: 'Utility', price: 150000, rent: 0, group: 'Utility' },
    { id: 13, name: "Hubungan Internasional", type: 'Property', price: 140000, rent: 10000, group: 'Pink' },
    { id: 14, name: "Sosiologi", type: 'Property', price: 160000, rent: 12000, group: 'Pink' },
    { id: 15, name: "Bikun", type: 'Railroad', price: 200000, rent: 25000, group: 'Railroad' },
    { id: 16, name: "Hukum Perdata", type: 'Property', price: 180000, rent: 14000, group: 'Orange' },
    { id: 17, name: "BEM", type: 'Chest', group: 'Chest' },
    { id: 18, name: "Hukum Pidana", type: 'Property', price: 180000, rent: 14000, group: 'Orange' },
    { id: 19, name: "Hukum Tata Negara", type: 'Property', price: 200000, rent: 16000, group: 'Orange' },

    // Top Row (20-30)
    { id: 20, name: "Pusgiwa", type: 'Corner', group: 'Special' },
    { id: 21, name: "Akuntansi", type: 'Property', price: 220000, rent: 18000, group: 'Red' },
    { id: 22, name: "SIAK-NG", type: 'Chance', group: 'Chance' },
    { id: 23, name: "Manajemen", type: 'Property', price: 220000, rent: 18000, group: 'Red' },
    { id: 24, name: "Ilmu Ekonomi", type: 'Property', price: 240000, rent: 20000, group: 'Red' },
    { id: 25, name: "Gerbang Utama", type: 'Railroad', price: 200000, rent: 25000, group: 'Railroad' },
    { id: 26, name: "Ilmu Komputer", type: 'Property', price: 260000, rent: 22000, group: 'Yellow' },
    { id: 27, name: "Sistem Informasi", type: 'Property', price: 260000, rent: 22000, group: 'Yellow' },
    { id: 28, name: "Danau UI", type: 'Utility', price: 150000, rent: 0, group: 'Utility' },
    { id: 29, name: "Teknologi Informasi", type: 'Property', price: 280000, rent: 24000, group: 'Yellow' },
    { id: 30, name: "Sanksi Akademik", type: 'Corner', group: 'Special' },

    // Right Column (31-39)
    { id: 31, name: "Teknik Sipil", type: 'Property', price: 300000, rent: 26000, group: 'Green' },
    { id: 32, name: "Teknik Elektro", type: 'Property', price: 300000, rent: 26000, group: 'Green' },
    { id: 33, name: "BEM", type: 'Chest', group: 'Chest' },
    { id: 34, name: "Teknik Mesin", type: 'Property', price: 320000, rent: 28000, group: 'Green' },
    { id: 35, name: "Balairung", type: 'Railroad', price: 200000, rent: 25000, group: 'Railroad' },
    { id: 36, name: "SIAK-NG", type: 'Chance', group: 'Chance' },
    { id: 37, name: "Kedokteran Gigi", type: 'Property', price: 350000, rent: 35000, group: 'DarkBlue' },
    { id: 38, name: "Biaya Praktikum", type: 'Tax', price: 100000, group: 'Tax' },
    { id: 39, name: "Kedokteran", type: 'Property', price: 400000, rent: 50000, group: 'DarkBlue' },
];

export const getTile = (id: number): TileInfo | undefined => {
    return TILES.find(t => t.id === id);
};

// Helper to get only buyable properties
export const getProperty = (id: number): TileInfo | undefined => {
    const tile = getTile(id);
    if (tile && (tile.type === 'Property' || tile.type === 'Railroad' || tile.type === 'Utility')) {
        return tile;
    }
    return undefined;
};

export const getGroupColor = (group: string) => {
    switch (group) {
        case 'Brown': return 'bg-yellow-900';
        case 'LightBlue': return 'bg-blue-200';
        case 'Pink': return 'bg-pink-400';
        case 'Orange': return 'bg-orange-400';
        case 'Red': return 'bg-red-500';
        case 'Yellow': return 'bg-yellow-400';
        case 'Green': return 'bg-green-500';
        case 'DarkBlue': return 'bg-blue-800';
        case 'Railroad': return 'bg-gray-800 text-white'; // Railroads are usually black/dark
        case 'Utility': return 'bg-gray-300'; // Utilities light gray
        case 'Tax': return 'bg-gray-400';
        case 'Chance': return 'bg-orange-200'; // Chance usually orange/question mark
        case 'Chest': return 'bg-blue-200'; // Community chest usually blue
        case 'Special': return 'bg-white'; // Corners
        default: return 'bg-gray-200';
    }
};

File: backend/src/game/board.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub enum PropertyGroup {
    Brown,
    LightBlue,
    Pink,
    Orange,
    Red,
    Yellow,
    Green,
    DarkBlue,
    Railroad,
    Utility,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PropertyInfo {
    pub id: usize,
    pub name: &'static str,
    pub price: i32,
    pub rent: i32, // Base rent for now
    pub group: PropertyGroup,
}

pub const PROPERTIES: &[PropertyInfo] = &[
    // Brown
    PropertyInfo { id: 1, name: "Matematika", price: 60_000, rent: 2_000, group: PropertyGroup::Brown },
    PropertyInfo { id: 3, name: "Fisika", price: 60_000, rent: 4_000, group: PropertyGroup::Brown },
    
    // Railroads
    PropertyInfo { id: 5, name: "Stasiun UI", price: 200_000, rent: 25_000, group: PropertyGroup::Railroad },
    PropertyInfo { id: 15, name: "Bikun", price: 200_000, rent: 25_000, group: PropertyGroup::Railroad },
    PropertyInfo { id: 25, name: "Gerbang Utama", price: 200_000, rent: 25_000, group: PropertyGroup::Railroad },
    PropertyInfo { id: 35, name: "Balairung", price: 200_000, rent: 25_000, group: PropertyGroup::Railroad },

    // Light Blue
    PropertyInfo { id: 6, name: "Sastra Inggris", price: 100_000, rent: 6_000, group: PropertyGroup::LightBlue },
    PropertyInfo { id: 8, name: "Arkeologi", price: 100_000, rent: 6_000, group: PropertyGroup::LightBlue },
    PropertyInfo { id: 9, name: "Filsafat", price: 120_000, rent: 8_000, group: PropertyGroup::LightBlue },

    // Pink
    PropertyInfo { id: 11, name: "Ilmu Komunikasi", price: 140_000, rent: 10_000, group: PropertyGroup::Pink },
    PropertyInfo { id: 13, name: "Hubungan Internasional", price: 140_000, rent: 10_000, group: PropertyGroup::Pink },
    PropertyInfo { id: 14, name: "Sosiologi", price: 160_000, rent: 12_000, group: PropertyGroup::Pink },

    // Utilities
    PropertyInfo { id: 12, name: "Perpustakaan UI", price: 150_000, rent: 0, group: PropertyGroup::Utility }, // Special rent logic
    PropertyInfo { id: 28, name: "Danau UI", price: 150_000, rent: 0, group: PropertyGroup::Utility },

    // Orange
    PropertyInfo { id: 16, name: "Hukum Perdata", price: 180_000, rent: 14_000, group: PropertyGroup::Orange },
    PropertyInfo { id: 18, name: "Hukum Pidana", price: 180_000, rent: 14_000, group: PropertyGroup::Orange },
    PropertyInfo { id: 19, name: "Hukum Tata Negara", price: 200_000, rent: 16_000, group: PropertyGroup::Orange },

    // Red
    PropertyInfo { id: 21, name: "Akuntansi", price: 220_000, rent: 18_000, group: PropertyGroup::Red },
    PropertyInfo { id: 23, name: "Manajemen", price: 220_000, rent: 18_000, group: PropertyGroup::Red },
    PropertyInfo { id: 24, name: "Ilmu Ekonomi", price: 240_000, rent: 20_000, group: PropertyGroup::Red },

    // Yellow
    PropertyInfo { id: 26, name: "Ilmu Komputer", price: 260_000, rent: 22_000, group: PropertyGroup::Yellow },
    PropertyInfo { id: 27, name: "Sistem Informasi", price: 260_000, rent: 22_000, group: PropertyGroup::Yellow },
    PropertyInfo { id: 29, name: "Teknologi Informasi", price: 280_000, rent: 24_000, group: PropertyGroup::Yellow },

    // Green
    PropertyInfo { id: 31, name: "Teknik Sipil", price: 300_000, rent: 26_000, group: PropertyGroup::Green },
    PropertyInfo { id: 32, name: "Teknik Elektro", price: 300_000, rent: 26_000, group: PropertyGroup::Green },
    PropertyInfo { id: 34, name: "Teknik Mesin", price: 320_000, rent: 28_000, group: PropertyGroup::Green },

    // Dark Blue
    PropertyInfo { id: 37, name: "Kedokteran Gigi", price: 350_000, rent: 35_000, group: PropertyGroup::DarkBlue },
    PropertyInfo { id: 39, name: "Kedokteran", price: 400_000, rent: 50_000, group: PropertyGroup::DarkBlue },
];

pub fn get_property(id: usize) -> Option<&'static PropertyInfo> {
    PROPERTIES.iter().find(|p| p.id == id)
}
